### ISSUE TITLE
ActionArg object cannot be interpreted as an integer

### ISSUE THREAD

**Issue by beheh** [jleclanche/fireplace#248]
For Tunnel Trogg we require constructs like `events = Overload(CONTROLLER).on(Buff(SELF, "LOE_018e") * Overload.AMOUNT)`. This leads to the following error:

```
TypeError: 'ActionArg' object cannot be interpreted as an integer
```

and the following call stack:

```
fireplace/card.py:290: in play
    self.game.queue_actions(self.controller, [actions.Play(self, target, index)])
fireplace/game.py:183: in queue_actions
    ret = self.trigger_actions(source, actions)
fireplace/game.py:209: in trigger_actions
    ret.append(action.trigger(source))
fireplace/actions.py:124: in trigger
    self.do(source, *args)
fireplace/actions.py:261: in do
    play_action()
fireplace/card.py:203: in action
    self.game.queue_actions(self, [actions.Overload(self.controller, self.overload)])
fireplace/game.py:183: in queue_actions
    ret = self.trigger_actions(source, actions)
fireplace/game.py:209: in trigger_actions
    ret.append(action.trigger(source))
fireplace/actions.py:124: in trigger
    self.do(source, *args)
fireplace/actions.py:303: in do
    self.broadcast(source, EventListener.ON, player, amount)
fireplace/actions.py:78: in broadcast
    self._broadcast(entity, source, at, *args)
fireplace/actions.py:74: in _broadcast
    entity.trigger_event(source, event, args)
fireplace/entity.py:84: in trigger_event
    source.game.queue_actions(self, actions)
fireplace/game.py:183: in queue_actions
    ret = self.trigger_actions(source, actions)
fireplace/game.py:209: in trigger_actions
    ret.append(action.trigger(source))
```

**Comment by beheh** [comment]
Once this is done we can merge https://github.com/beheh/fireplace/tree/tunnel_trogg.

**Comment by jleclanche** [comment]
See if this is fixed by e68c6d75931dab1fa28a59b82b5a43f1f8602630
